pyproject.toml

```diff
@@ -78,6 +77,10 @@ sphinx-autodoc-typehints = "^1.12.0"
 sphinx-rtd-theme = "^0.5.2"
 types-PyYAML = "^6.0.4"
 
+[tool.poetry.extras]
+feast = ["feast"]
+git = ["GitPython"]
```

Comment:
I'm wondering if we should use extra here or leverage the new group feature from poetry https://python-poetry.org/docs/master/managing-dependencies/#dependency-groups